### PR TITLE
Deprecate `coordinates.get_moon()`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

`get_moon()` is simply a shortcut to calling `get_body("moon")` with a docstring that tends to get out of sync from the docstring of `get_body()` (as evidenced by #12683), and the presence of which complicates unit tests in `astropy.coordinates.tests.test_solar_system` because Moon is treated differently from all the other Solar System bodies. Deprecating and eventually removing it simplifies the code without sacrificing functionality or readability. Note that this pull request removes a net ~22~ 23 lines of code despite adding a deprecation test and a change log entry and `get_moon()` hasn't even been removed yet.

It seems to me the reason a separate function was created for the Moon in the first place was because it was introduced in 4a59bb8885817547e40caf570e21ecf4816bcbc2 together with a function called `get_planet()`, and calling ~`get_planet("moon")`~ the equivalent of `get_planet("moon")` would have been confusing. But `get_planet()` was renamed to `get_body()` already in cd408c2744b6309d6c871819238fec7893e8fa7e (committed on the same day). I don't understand why `get_moon()` was not removed then.

### CODE REVIEW

**Comment by bmorris3** [comment]
Hi @eerovaher!

> But get_planet() was renamed to get_body() already in https://github.com/astropy/astropy/commit/cd408c2744b6309d6c871819238fec7893e8fa7e (committed on the same day). I don't understand why get_moon() was not removed then.

I implemented a convenience function called `get_moon`, even though it was only a tiny helper wrapper, in part to keep special parity with the [`get_sun`](https://docs.astropy.org/en/stable/api/astropy.coordinates.get_sun.html) method. 

A bit of history: my original `get_planet` implementation in astropy used indices to identify each solar system object ([discussion](https://github.com/astropy/astropy/pull/4746#discussion_r57761816)), and specifying the Moon as an integer (especially `3`) seemed confusing. 

Unlike `get_moon`, `get_sun` has a special right to exist independent of `get_body/get_moon` because it uses the `erfa` routines to get the solar position rather than `jplephem` like `get_body`:

https://github.com/astropy/astropy/blob/01d28392ff4c716609df5cf516d0da9771357153/astropy/coordinates/funcs.py#L150-L154

**TL;DR**: `get_moon` doesn't need to exist, and I agree that it's clearer if it does not exist!

File: docs/changes/coordinates/14354.api.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+``get_moon()`` is deprecated and may be removed in a future version of
+``astropy``. The geocentric coordinates of the Moon can be obtained by calling
+``get_body("moon")``.
```
Comment (bmorris3) [inline docs/changes/coordinates/14354.api.rst:3]:
```suggestion
``get_body("moon", time, [location])``.
```

**Review by bmorris3 — APPROVED** [review]
Looks great!

**Comment by StuartLittlefair** [comment]
I'm happy to approve this in principle, since there really is no need for `get_moon` to exist. 

However, it does make me question why we don't deprecate `get_sun` at the same time? @bmorris3 isn't right about one using `erfa` and the other using `jplephem`. `get_body` uses whatever the solar system ephemeris is set to, and the default is to use `erfa`'s  ephemeris.

Therefore, by default both `get_body` and `get_sun` use `erfa.epv00` to find the relative position of the Earth and the Sun. They differ in how the effects of aberration and the light travel time are handled. I'm not sure there's justification for having two different implementations of the same thing?

**Comment by eerovaher** [comment]
I've found that removing `get_sun()` in favor of `get_body("sun")` was already discussed in #4952. In https://github.com/astropy/astropy/pull/4952#issuecomment-220379031 @eteq said that `get_sun()` should not be removed, but I haven't been able to find what the justification for that statement was. I am somewhat hesitant about deprecating `get_sun()` before I understand why it wasn't done already at the time of #4952.

**Comment by StuartLittlefair** [comment]
Interesting, I’d like to hear @eteq ’s view on why get_sun should survive. 

Either way, given that there is definitely no reason to keep `get_moon` and the title of this PR is lunar only, I’ll approve this and open a different issue to propose deprecating `get_sun`

**Review by StuartLittlefair — APPROVED** [review]

File: docs/changes/coordinates/14354.api.rst (hunk 1 above)
Comment (eerovaher) [reply docs/changes/coordinates/14354.api.rst]: I've changed the phrasing of the change log entry, although I tried to make the example function calls less shell-like and more Python-like.

File: astropy/coordinates/tests/test_solar_system.py [hunk 2]
```diff
@@ -212,23 +213,6 @@ def test_de432s_planet(self, body):
...
-    @pytest.mark.skipif(not HAS_JPLEPHEM, reason="requires jplephem")
-    def test_de432s_moon(self):
-        astropy = get_moon(self.t, ephemeris="de432s")
-        horizons = self.horizons["moon"]
-
-        # convert to true equator and equinox
-        astropy = astropy.transform_to(self.apparent_frame)
-
-        # Assert sky coordinates are close.
-        assert astropy.separation(horizons) < de432s_separation_tolerance_moon
```
Comment (mhvk) [inline astropy/coordinates/tests/test_solar_system.py:225]: Note that the definition of `de432s_separation_tolerance_moon` can also be removed (and it is the same as for planets, so the tests indeed stay the same)

**Review by mhvk — COMMENTED** [review]
Agreed that `get_moon` can be deprecated. Just a small comment on another line that can be removed.

File: astropy/coordinates/tests/test_solar_system.py (hunk 2 above)
Comment (eerovaher) [reply astropy/coordinates/tests/test_solar_system.py:225]: I've removed `de432s_separation_tolerance_moon`.

Comment (eerovaher) [reply astropy/coordinates/tests/test_solar_system.py:225]: @mhvk, I've implemented your suggestion. Can this be merged now?
